WebView/app/src/main/assets/index.html

```diff
@@ -16,8 +16,26 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <link href="app.css" rel="stylesheet">
     <script async src="main.js"></script>
+    <meta name="color-scheme" content="dark light">
+    <script>
+      // If dark mode is not supported then load light.css
+      if (window.matchMedia('(prefers-color-scheme: dark)').media === 'not all') {
+      document.head.insertAdjacentHTML(
+        'beforeend',
+        '<link rel="stylesheet" href="light.css" onload="document.documentElement.style.display = \'\'">'
+       );
+     }
+    </script>
+    <!--
+      If dark mode is supported, conditionally either load the light or the dark stylesheet. If
+      `prefers-color-scheme` is not supported the media query is unknown and the files are loaded
+       with `lowest` priority.
+    -->
```

Comment:
Is there a reason we can't merge these into the same style sheet? Something like:


```css
// Apply light styles for the default theme:
h1 {
  color: white;
}

// (not sure if this is the right CSS syntax) If we're in dark theme, override these colors:
if (prefers-color-scheme: dark) {
  h1 {
    color: black;
  }
}
```

Reply:
Yes, anything outside of a designated theme has preference over values set inside a theme. It only works to have the themes defined in the same file if they were both clearly marked as either light or dark with 'prefers-color-scheme'. Once we remove 'prefers-color-scheme:light' from around the light theme (for those cases in which prefers-color-scheme is not supported) the light theme attributes always override the dark theme because they become part of what is considered universal. Additionally, it allows us optimize load speed by not downloading CSS for the alternate mode in the critical rendering path. More in depth explanation [here](https://web.dev/defer-non-critical-css/).

Reply:
Splitting CSS unfortunately seems very error prone. For example, your `light.css` [centers text](https://github.com/gcoleman799/views-widgets-samples/blob/60105b6edf386bf9dc9f1d60c7befda87b2bd5e4/WebView/app/src/main/assets/light.css#L6) but `dark.css` is [missing this](https://github.com/gcoleman799/views-widgets-samples/blob/60105b6edf386bf9dc9f1d60c7befda87b2bd5e4/WebView/app/src/main/assets/dark.css#L1). I don't think it's worth the benefit to page load time (we don't have much CSS).

https://stackoverflow.com/a/38593143 suggests you can write something like this:

```css
/* Declare a default theme first. This has the lowest precedence. */
p {
  color: blue;
}
/*
Declare dark/light themes next. This ensures they will override the default
theme. In our case, we can probably just let the default theme be our light
theme.
*/
@media (prefers-color-scheme: dark) {
  p {
     color: green;
  }
}
@media (prefers-color-scheme: light) {
  p {
     color: red;
  }
}
```

This works in my Chrome on Mac (I'm still on 83.0.4103.116), but it seems like this could plausibly work correctly on older WebView versions too.